src/datasets/fingerprint.py

```diff
@@ -34,29 +56,31 @@ def __init__(self):
         self.m = xxhash.xxh64()
 
     @classmethod
-    def hash_bytes(cls, value):
+    def hash_bytes(cls, value: Union[bytes, List[bytes]]) -> str:
         value = [value] if isinstance(value, bytes) else value
         m = xxhash.xxh64()
         for x in value:
             m.update(x)
         return m.hexdigest()
 
     @classmethod
-    def hash_default(cls, value):
+    def hash_default(cls, value: Any) -> str:
         return cls.hash_bytes(dumps(value))
 
     @classmethod
-    def hash(cls, value):
+    def hash(cls, value: Any) -> str:
         if type(value) in cls.dispatch:
             return cls.dispatch[type(value)](cls, value)
         else:
             return cls.hash_default(value)
 
-    def update(self, value):
-        self.m.update(f"=={type(value)}==".encode("utf8"))
-        self.m.update(self.hash(value).encode("utf-8"))
+    def update(self, value: Any) -> None:
+        header_for_update = f"=={type(value)}=="
+        value_for_update = self.hash(value)
+        self.m.update(header_for_update.encode("utf8"))
+        self.m.update(value_for_update.encode("utf-8"))
```

Comment:
I changed the order here to not update the hash in case `self.hash` fails (for unpicklable objects for example).

Reply:
@lhoestq

Reply:
Good point!